// react-native-pitchy/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
const LINKING_ERROR = `The package 'react-native-pitchy' doesn't seem to be linked. Make sure: \n\n` + _reactNative.Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo Go\n';
const PitchyNativeModule = _reactNative.NativeModules.Pitchy ? _reactNative.NativeModules.Pitchy : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
const eventEmitter = new _reactNative.NativeEventEmitter(PitchyNativeModule);
const Pitchy = {
  init(config) {
    return PitchyNativeModule.init({
      bufferSize: 4096,
      minVolume: -60,
      algorithm: 'ACF2+',
      ...config
    });
  },
  start() {
    return PitchyNativeModule.start();
  },
  stop() {
    return PitchyNativeModule.stop();
  },
  isRecording() {
    return PitchyNativeModule.isRecording();
  },
  addListener(callback) {
    return eventEmitter.addListener('onPitchDetected', callback);
  }
};
var _default = exports.default = Pitchy;
