// react-native-pitchy/lib/module/index.js
"use strict";

import { NativeModules, NativeEventEmitter, Platform } from 'react-native';
const LINKING_ERROR = `The package 'react-native-pitchy' doesn't seem to be linked. Make sure: \n\n` + Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo Go\n';
const PitchyNativeModule = NativeModules.Pitchy ? NativeModules.Pitchy : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
const eventEmitter = new NativeEventEmitter(PitchyNativeModule);
const Pitchy = {
  init(config) {
    return PitchyNativeModule.init({
      bufferSize: 4096,
      minVolume: -60,
      algorithm: 'ACF2+',
      ...config
    });
  },
  start() {
    return PitchyNativeModule.start();
  },
  stop() {
    return PitchyNativeModule.stop();
  },
  isRecording() {
    return PitchyNativeModule.isRecording();
  },
  addListener(callback) {
    return eventEmitter.addListener('onPitchDetected', callback);
  }
};
export default Pitchy;
